packages/answers-kit/src/modules/faculties/index.ts

```diff
@@ -0,0 +1,155 @@
+import type { Faculty, Subject } from '@/types.js';
+import { getDefaultHeaders } from '@/utils/headers.js';
+import { handleResult } from '@/utils/result.js';
+import type {
+  CreateFacultyArgs,
+  DeleteFacultyArgs,
+  GetFacultySubjectsArgs,
+  IFacultiesModule,
+  UpdateFacultyArgs,
+} from './types.js';
+
+export class FacultiesModule implements IFacultiesModule {
+  private readonly url: string;
+
+  constructor(client: string) {
+    this.url = client;
+  }
+
+  /**
+   * Find all faculties
+   *
+   * @example Example usage:
+   * ```ts
+   * const faculty = await answersKit.faculties.findMany()
+   * ```
+   *
+   * @returns an array of faculty objects
+   *
+   * @publicApi
+   * */
+
+  async findMany(): Promise<Faculty[]> {
+    const response = await fetch(`${this.url}/faculties`);
+
+    return response.json();
+  }
+
+  /**
+   * Get faculty subjects
+   *
+   * @example Example usage:
+   * ```ts
+   * const subjects = await answersKit.faculties.getFacultySubjects({
+   *   id: 1
+   * })
+   * ```
+   *
+   * @returns an array of subject objects
+   *
+   * @publicApi
+   * */
+
+  async getFacultySubjects({ id }: GetFacultySubjectsArgs): Promise<Subject[]> {
```

Comment:
I guess there is no sense in passing faculty `id` as object field. Since this method will not be extended in future, it would be better to get rid of `GetFacultySubjectsArgs` interface and pass `id` just as numeric argument